webapp/views.py

```diff
@@ -1473,9 +1473,57 @@ def serve_sitemap():
 def process_local_communities(local_communities):
     def display_local_communities():
         metadata_table = local_communities.get_category_index_metadata("locos")
+
+        # Group communities by continent
+        valid_continents = [
+            "africa",
+            "asia",
+            "europe",
+            "north america",
+            "south america",
+            "oceania",
+        ]
+        communities_by_continent = {}
+        for community in metadata_table:
+            continent = community.get("continent")
+            if continent and continent.lower() in valid_continents:
+                if continent not in communities_by_continent:
+                    communities_by_continent[continent] = []
+                communities_by_continent[continent].append(community)
+        communities_by_continent = dict(
+            sorted(communities_by_continent.items())
+        )
+
```

Comment:
What I could deduce from code is that if the community isn't part of valid continent we will not show it in the tabs, but if it has valid co-ordinates we will show it in the map.
So if we have a new community which might not have a valid continent, it will still show up in the map.
Not sure if we should do something about this, just putting in thoughts.

Also while we are at it, we can create markers and community list with in a single loop.

Reply:
I'll answer in two messages for clarity, for the first part
Let me explain what the code is doing:
- It checks for the presence of the 'continent' key/value
- If it is a valid continent (one of the 7 in the list) we check if that continent is already in the object and add it if not. 
- We then add that community into that continent 

This means if a continent doesn't have any communities, it wont be displayed. As for the 'valid continent', I don't think I understand. There is only 7 possible continents and they are all covered here. The content is also controlled by someone with Canonical.

Reply:
The loops cannot be combined as one loop creates a dict and the other creates a list. This is intentional as it simplifies the rendering in the template.

Reply:
For the continent part, what I meant was that people could omit continent, name it Australia(the list has "oceania") hence keep markers for the communities that have valid continent.
But again it's a low percentage edge case.

Reply:
The metadata table that the map pulls from is maintained by someone internal and they are aware of that continent needs to be included, so this shouldn't be an issue